File: 2639-separate-the-digits-in-an-array/2639-separate-the-digits-in-an-array.java
import java.util.*;
import java.io.*;

class Solution {
    
    public static ArrayList splitNumber(int number) {
        ArrayList<Integer> digits = new ArrayList<Integer>();
        while(number != 0) {
            digits.add(number%10);
            number /= 10;
        }
        Collections.reverse(digits);
        return digits;
    }
    
    public int[] separateDigits(int[] nums) {
	    ArrayList<Integer> output = new ArrayList<Integer>();
		for(int i=0; i<nums.length; i++) {
		    output.addAll(splitNumber(nums[i]));
		}
        
        int[] finalArray = new int[output.size()];
        int idx = 0;
        for(Integer I : output)
        {
            finalArray[idx++] = (int)I;
        }
        return finalArray;
    }
}

File: 2639-separate-the-digits-in-an-array/2639-separate-the-digits-in-an-array.py
class Solution:
    def num_to_digit(self, num:int) -> List[int]:
        digit_list = []
        while(num > 0):
            digit_list.append(num%10)
            num = num//10
        return digit_list[:: -1]

    def separateDigits(self, nums: List[int]) -> List[int]:        
        # output = []
        # for ele in nums:
        #     output.extend(self.num_to_digit(ele))
        # return output
        output = []
        for num in nums:
            output.extend([int(x) for x in str(num)])
            # output.extend([num if len(str(num))==1 else int(x) for x in str(num)])
        return output

    # def digits(self, num):
    #     dig = []
    #     while(num !=0):
    #         dig.append(num%10)
    #         num//=10
    #     return dig
    # def separateDigits(self, nums: List[int]) -> List[int]: 
    #     output = []
    #     for ele in nums[::-1]:
    #         output.extend(self.digits(ele))
    #     return(output[::-1])